docs/Extensions.md

```diff
@@ -0,0 +1,50 @@
+# fastlane extensions
+
+### Local actions
+
+You can create your own actions to extend the functionality of `fastlane`. The action you create will behave exactly like the built in actions.
+
+Just run `fastlane new_action`. Then enter the name of the action and edit the generated Ruby file in `fastlane/actions/[action_name].rb`.
+
+From then on, you can just start using your action in your `Fastfile`.
+
+### Submitting the action to the fastlane main repo
+
+If you think your extension will be useful to other developers as well, clone fastlane, copy your local action to `fastlane/lib/fastlane/actions` and submit a pull request.
```

Comment:
@jcampbell05 exactly. I've seen it all happen in the jenkins/hudson communities, where some plugins were originally bundled together with jenkins, and later, work was made to detach them from the core bundle.